cscs-checks/apps/spark/spark_check.py

```diff
@@ -39,16 +35,44 @@ def prepare_run(self):
             exec_cores = 9
 
         self.variables = {
-            'SPARK_WORKER_CORES': '%s' % num_workers,
+            'SPARK_WORKER_CORES': f'{num_workers}',
             'SPARK_LOCAL_DIRS': '"/tmp"',
         }
-        self.executable = (
-            f'spark-submit --conf spark.default.parallelism={num_workers} '
-            f'--conf spark.executor.cores={exec_cores} '
-            f'--conf spark.executor.memory=15g --master $SPARKURL '
-            f'--class org.apache.spark.examples.SparkPi '
-            f'$EBROOTSPARK/examples/jars/spark-examples*.jar 10000;'
-        )
+        self.executable = 'spark-submit'
+        self.executable_opts = [
+            f'--conf spark.default.parallelism={num_workers}',
+            f'--conf spark.executor.cores={exec_cores}',
+            f'--conf spark.executor.memory=15g',
+            f'--master $SPARKURL'
+        ]
         # The job launcher has to be changed since the `spark-submit`
         # script is not used with srun.
         self.job.launcher = getlauncher('local')()
+
+
+@rfm.simple_test
+class SparkCheck(SparkBaseCheck):
+    def __init__(self):
+        super().__init__()
+        self.descr = 'Simple calculation of pi with Spark'
+        self.sourcesdir = None
+
+    @rfm.run_before('run')
+    def prepare_run(self):
+        super().prepare_run()
+        self.executable_opts.extend([
+            '--class org.apache.spark.examples.SparkPi',
+            '$EBROOTSPARK/examples/jars/spark-examples*.jar 10000'
+        ])
+
+
+@rfm.simple_test
+class PySparkCheck(SparkBaseCheck):
```

Comment:
I think it's better if you use a single test and parametrize it. The two tests are almost identical.